### PR TITLE
ENH: Multiple filename support in `.load`

### PR DESCRIPTION
**Pull request by rstoneback**
Adds support for dealing with multiple filenames across instruments. Required for loading a range of dates from pysat.

```
In [1]: import pysat                                                                                                                                                                                                        

In [2]: f107 = pysat.Instrument('sw', 'f107')                                                                                                                                                                               

In [3]: f107.load()                                                                                                                                                                                                         

In [4]: f107.data                                                                                                                                                                                                           
Out[4]: 
             f107
2015-01-01  132.9
2015-01-02  141.0
2015-01-03  143.8
2015-01-04  144.8
2015-01-05  137.2
...           ...
2015-08-27  112.5
2015-08-28  111.3
2015-08-29  102.1
2015-08-30   93.4
2015-08-31   92.7

[242 rows x 1 columns]
```

```
In [9]: kp = pysat.Instrument('sw', 'kp', update_files=True)                                                                                                                                                                

In [11]: kp.download(dt.datetime(2001, 1, 1), dt.datetime(2001, 12, 31))                                                                                                                                                    

In [12]: kp.load()                                                                                                                                                                                                          

In [13]: kp.data                                                                                                                                                                                                            
Out[13]: 
                           Kp
time                         
2001-01-01 00:00:00  0.000000
2001-01-01 03:00:00  0.333333
2001-01-01 06:00:00  1.000000
2001-01-01 09:00:00  1.000000
2001-01-01 12:00:00  0.333333
...                       ...
2001-12-31 09:00:00  1.333333
2001-12-31 12:00:00  1.333333
2001-12-31 15:00:00  3.000000
2001-12-31 18:00:00  3.666667
2001-12-31 21:00:00  4.000000

[2920 rows x 1 columns]
```

Dst downloads weren't working for me.

### CODE REVIEW

File: pysatSpaceWeather/instruments/sw_f107.py [hunk 1]
```diff
@@ -124,27 +124,31 @@ def load(fnames, tag=None, sat_id=None):
...
+        elif tag == 'all':
+            result = pds.read_csv(fname, index_col=0, parse_dates=True)
+        elif tag == 'daily' or tag == 'prelim':
+            result = pds.read_csv(fname, index_col=0, parse_dates=True)
+        elif tag == 'forecast':
+            # load forecast data
+            result = pds.read_csv(fname, index_col=0, parse_dates=True)
+        elif tag == '45day':
+            # load forecast data
+            result = pds.read_csv(fname, index_col=0, parse_dates=True)
```
Comment (jklenzing) [inline pysatSpaceWeather/instruments/sw_f107.py:148]:
```suggestion
        elif tag in ['all', 'daily', 'prelim', 'forecast', '45day']:
            result = pds.read_csv(fname, index_col=0, parse_dates=True)
```

**Review by jklenzing — APPROVED** [review]
This looks good.  You'll need to merge in the latest from develop for Travis to work.

**Comment by jklenzing** [comment]
Also, I just noticed that this passed on Travis 2 days ago, without signs of #9 or #10.  Not sure what's changed in the past two days.  The relevant bits of code that are failing have not changed.

**Review by jklenzing — CHANGES_REQUESTED** [review]
Tried out the dst locally.  Downloads work, but load does not.  Pulling back approval for now.

**Comment by jklenzing** [comment]
Copying the above methods
```
import datetime as dt
import pysat  
import pysatSpaceWeather
dst = pysat.Instrument(inst_module=pysatSpaceWeather.instruments.sw_dst)
dst.download(dt.datetime(2001, 1, 1), dt.datetime(2001, 12, 31)) 
dst.load()
```
yields
```
TypeError                                 Traceback (most recent call last)
<ipython-input-8-63527e78655e> in <module>
----> 1 dst.load()

~/code/core/pysat/pysat/_instrument.py in load(self, yr, doy, date, fname, fid, verifyPad)
   1425             estr = 'Must supply a yr,doy pair, or datetime object, or filename'
   1426             estr = '{:s} to load data from.'.format(estr)
-> 1427             raise TypeError(estr)
   1428 
   1429         self.orbits._reset()

TypeError: Must supply a yr,doy pair, or datetime object, or filename to load data from.
```

**Comment by rstoneback** [comment]
> ~/code/core/pysat/pysat/_instrument.py in load(self, yr, doy, date, fname, fid, verifyPad)
>    1425             estr = 'Must supply a yr,doy pair, or datetime object, or filename'
>    1426             estr = '{:s} to load data from.'.format(estr)
> -> 1427             raise TypeError(estr)
>    1428 
>    1429         self.orbits._reset()
> 
> TypeError: Must supply a yr,doy pair, or datetime object, or filename to load data from.
> ```

I forgot to mention this requires the `load_data_range` branch from pysat. Trying to ensure support is in all downstream packages before making the pull. I closed the existing pull for the branch because it was already too crowded with comments.

**Comment by jklenzing** [comment]
That works.  Tracking down the sources of #9 and #10, since these are popping up here now.

**Comment by rstoneback** [comment]
Are you also sure the Dst download worked (have to check at filesystem level)? It isn't for me.

**Comment by jklenzing** [comment]
> Are you also sure the Dst download worked (have to check at filesystem level)? It isn't for me.

It's a new system, so I didn't have any dst data yet.  Outside the scope here, but since the data is yearly, it grabbed the same file 365 times.  Something for pysat/pysat#58?

**Comment by jklenzing** [comment]
`dst.load()` is working for me if I use the load_date_range branch is pysat

**Comment by rstoneback** [comment]
I improved the download routine to only download by unique year.

**Comment by rstoneback** [comment]
Test failures are related to #9 and #10, unrelated to changes in this pull. My intended changes are complete.

**Comment by rstoneback** [comment]
> Test failures are related to #9 and #10, unrelated to changes in this pull. My intended changes are complete.

Ready for re-review.

**Review by aburrell — APPROVED** [review]

**Pull request merged**